app/src/main/java/org/fossasia/pslab/fragment/HalfwaveRectifierFragment.java

```diff
@@ -0,0 +1,150 @@
+
+        ArrayAdapter<String> rangesAdapter;
+        rangesAdapter = new ArrayAdapter<String>(this.getActivity(), R.layout.custom_spinner, ranges);
+        rangesAdapter.setDropDownViewResource(android.R.layout.simple_dropdown_item_1line);
+        spinnerRangeCh1.setAdapter(rangesAdapter);
+
+        spinnerRangeCh1.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                switch (position) {
+                    case 0:
+                        ((OscilloscopeActivity) getActivity()).setLeftYAxisScale(16, -16);
+                        ((OscilloscopeActivity) getActivity()).setRightYAxisScale(16, -16);
+                        break;
+                    case 1:
+                        ((OscilloscopeActivity) getActivity()).setLeftYAxisScale(8, -8);
+                        ((OscilloscopeActivity) getActivity()).setRightYAxisScale(8, -8);
+                        break;
+                    case 2:
+                        ((OscilloscopeActivity) getActivity()).setLeftYAxisScale(4, -4);
+                        ((OscilloscopeActivity) getActivity()).setRightYAxisScale(4, -4);
+                        break;
+                    case 3:
+                        ((OscilloscopeActivity) getActivity()).setLeftYAxisScale(3, -3);
+                        ((OscilloscopeActivity) getActivity()).setRightYAxisScale(3, -3);
+                        break;
+                    case 4:
+                        ((OscilloscopeActivity) getActivity()).setLeftYAxisScale(2, -2);
+                        ((OscilloscopeActivity) getActivity()).setRightYAxisScale(2, -2);
+                        break;
+                    case 5:
+                        ((OscilloscopeActivity) getActivity()).setLeftYAxisScale(1.5, -1.5);
+                        ((OscilloscopeActivity) getActivity()).setRightYAxisScale(1.5, -1.5);
+                        break;
+                    case 6:
+                        ((OscilloscopeActivity) getActivity()).setLeftYAxisScale(1, -1);
+                        ((OscilloscopeActivity) getActivity()).setRightYAxisScale(1, -1);
+                        break;
+                    case 7:
+                        ((OscilloscopeActivity) getActivity()).setLeftYAxisScale(500, -500);
+                        ((OscilloscopeActivity) getActivity()).setRightYAxisScale(500, -500);
+                        break;
+                    case 8:
+                        ((OscilloscopeActivity) getActivity()).setLeftYAxisScale(160, -160);
+                        ((OscilloscopeActivity) getActivity()).setRightYAxisScale(160, -160);
+                        break;
+                }
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> parent) {
+
+            }
+        });
+
+        return v;
+    }
+
+    public void onButtonPressed(Uri uri) {
```

Comment:
I think, there is quite unused code in fragments like ```onFragmentInteractionListener``` interface and params putting in bundle, etc.
Could you tell if it has some use?